internal/status.go

```diff
@@ -0,0 +1,41 @@
+package internal
+
+import (
+	"context"
+	"fmt"
+	"github.com/algorand/go-algorand-sdk/client/v2/algod"
+)
+
+// StatusModel represents a status response from algod.Status
+type StatusModel struct {
+	HeartBeat chan uint64 // Subscription Channel
+	LastRound uint64      // Last recorded round
+}
+
+// String prints the last round value
+func (m *StatusModel) String() string {
+	return fmt.Sprintf("Last round: %d", m.LastRound)
+}
+
+// Fetch handles algod.Status
+func (m *StatusModel) Fetch(algodClient *algod.Client) error {
+	s, err := algodClient.Status().Do(context.Background())
+	if err != nil {
+		return err
+	}
+	m.LastRound = s.LastRound
+	return nil
+}
+
+// Watch uses algod.StatusAfterBlock to wait for changes and emits to the HeartBeat channel
+func (m *StatusModel) Watch(ctx context.Context, algodClient *algod.Client) error {
+	lastRound := uint64(0)
```

Comment:
why do we start from 0?

Reply:
Good catch, it's missing it's update from the status. I will update the last round counter